core/offchain/src/lib.rs

```diff
@@ -157,16 +157,54 @@ impl<Client, Storage, KeyProvider, Block> OffchainWorkers<
 				at.clone(),
 				network_state.clone(),
 			);
-			debug!("Running offchain workers at {:?}", at);
-			let api = Box::new(api);
-			runtime.offchain_worker_with_context(&at, ExecutionContext::OffchainWorker(api), *number).unwrap();
+			debug!("Spawning offchain workers at {:?}", at);
+			let number = *number;
+			let client = self.client.clone();
+			spawn_worker(move || {
+				let runtime = client.runtime_api();
+				let api = Box::new(api);
+				debug!("Running offchain workers at {:?}", at);
+				let run = runtime.offchain_worker_with_context(
+					&at,
+					ExecutionContext::OffchainWorker(api),
+					number
+				);
+				if let Err(e) =	run {
+					log::error!("Error running offchain workers at {:?}: {:?}", at, e);
+				}
+			});
 			futures::future::Either::A(runner.process())
```

Comment:
What is process() returning?

Reply:
A future in which we process APIs that has to be invoked asynchronously - it's `submit_transaction` currently, since we can't recursively call into the runtime.